Add gallery of examples to documentation (#188)
Happy-Algorithms-League/hal-cgp: examples/README.rst

========
Examples
========

> This folder contains a number of example scripts that demonstrate the different use cases of `hal-cgp`.

Reviewer: ```suggestion
This folder contains a number of example scripts that demonstrate some use cases of `hal-cgp`.
```

Author: thanks, I used "several" in the end.
